From dd11293e849a30ff0a72b1cfed804f238ef536cb Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Wed, 8 May 2013 18:09:54 +0200
Subject: [PATCH 40/40] BUG/MINOR: acl: fix a double free during exit when
 using PCRE_JIT

When freeing ACL regex, we don't want to perform the free() in regex_free()
as it's already performed in free_pattern(). The double free only happens
when using PCRE_JIT when freeing everything during exit so it's harmless
but exhibits libc errors during a reload/restart.

Bug reported by Seri.
---
 include/common/regex.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/include/common/regex.h b/include/common/regex.h
index bab1a55..0104019 100644
--- a/include/common/regex.h
+++ b/include/common/regex.h
@@ -79,7 +79,6 @@ static inline void regex_free(regex *preg) {
 #ifdef USE_PCRE_JIT
 	pcre_free_study(preg->extra);
 	pcre_free(preg->reg);
-	free(preg);
 #else
 	regfree(preg);
 #endif
-- 
1.8.1.5

